### DISCUSSION TITLE
Why does it need root permissions for installing homebrew?

### DISCUSSION THREAD

**Discussion by codingbutstillalive** [Homebrew/discussions#609] [Everyday usage]
I wonder why it is necessary to grant root permission when installing homebrew.

That would allow all kinds of stuff to be silently installed, including root kits. All it needs is a hack of the Git repositories. See Solarwind, e.g.

**Comment by Linus-XZX** [comment]
Not an American but understand your concerns :)
Homebrew is installed in `/usr/local` (for my Intel MBP at least), and this directory is 755 `root:wheel` starting macOS 11, and even in earlier versions installing here is quite complicated regarding user privileges.
Besides, Homebrew installation may call `xcode-select` to initiate installation of a matching version of CLT, which requires root privileges.
For your concerns, maybe you can try reading the installation script at the official site. Or more drastically you can set `sudo` timeout to be 0, so that you may know for sure how many times root privileges are requested (at the cost of typing your password every time).

**Comment by SMillerDev** [comment]
> I wonder why it is necessary to grant root permission when installing homebrew.

Because homebrew needs to create directories in your system for the default install. You can always manually install homebrew if you want more control over sudo usage.

> That would allow all kinds of stuff to be silently installed, including root kits. All it needs is a hack of the Git repositories. See Solarwind, e.g.

Running any kind of software would allow stuff to be silently installed. What if I write a script that prepends a malicious sudo binary to your path using `~/.profile`? No sudo required. In the end you're going to have to trust your package manager if you want to use it.
